cloudstack/resource_cloudstack_project.go

```diff
@@ -0,0 +1,537 @@
+		if len(project.Owner) > 0 {
+			foundAccountID := false
+			for _, owner := range project.Owner {
+				if accountid, ok := owner["accountid"]; ok {
+					d.Set("accountid", accountid)
+					foundAccountID = true
+					break
+				}
+			}
+			if !foundAccountID {
+				log.Printf("[DEBUG] Project %s owner information doesn't contain accountid, keeping original value", d.Id())
+			}
+		} else {
+			log.Printf("[DEBUG] Project %s owner information not available yet, keeping original accountid value", d.Id())
+		}
+	}
+
+	if _, ok := d.GetOk("userid"); ok {
+		if len(project.Owner) > 0 {
+			foundUserID := false
+			for _, owner := range project.Owner {
+				if userid, ok := owner["userid"]; ok {
+					d.Set("userid", userid)
+					foundUserID = true
+					break
+				}
+			}
+			if !foundUserID {
+				log.Printf("[DEBUG] Project %s owner information doesn't contain userid, keeping original value", d.Id())
+			}
+		} else {
+			log.Printf("[DEBUG] Project %s owner information not available yet, keeping original userid value", d.Id())
+		}
+	}
+
+	return nil
+}
+
+func resourceCloudStackProjectUpdate(d *schema.ResourceData, meta any) error {
+	cs := meta.(*cloudstack.CloudStackClient)
+
+	// Check if the name or display text is changed
+	if d.HasChange("name") || d.HasChange("display_text") {
+		// Create a new parameter struct
+		p := cs.Project.NewUpdateProjectParams(d.Id())
+
+		// Set the name and display_text if they have changed
+		// Note: The 'name' parameter is only available in API 4.19 and higher
+		// If you're using API 4.18 or lower, the SetName method might not work
+		// In that case, you might need to update the display_text only
+		if d.HasChange("name") {
+			p.SetName(d.Get("name").(string))
+		}
+
+		if d.HasChange("display_text") {
+			p.SetDisplaytext(d.Get("display_text").(string))
+		}
+
+		log.Printf("[DEBUG] Updating project %s", d.Id())
+		_, err := cs.Project.UpdateProject(p)
+		if err != nil {
+			return fmt.Errorf("Error updating project %s: %s", d.Id(), err)
+		}
+	}
+
+	// Check if the account, accountid, or userid is changed
+	if d.HasChange("account") || d.HasChange("accountid") || d.HasChange("userid") {
+		// Create a new parameter struct
+		p := cs.Project.NewUpdateProjectParams(d.Id())
+
+		// Set swapowner to true to swap ownership with the account/user provided
+		p.SetSwapowner(true)
+
+		// Set the account if it has changed
+		if d.HasChange("account") {
+			p.SetAccount(d.Get("account").(string))
+		}
+
+		// Set the userid if it has changed
+		if d.HasChange("userid") {
+			p.SetUserid(d.Get("userid").(string))
+		}
+
+		// Note: accountid is not directly supported by the UpdateProject API,
+		// but we can use the account parameter instead if accountid has changed
+		if d.HasChange("accountid") && !d.HasChange("account") {
+			// If accountid has changed but account hasn't, we need to look up the account name
+			// This is a placeholder - in a real implementation, you would need to look up
```

Comment:
@ianc769 Can we implement this? The changes should be small enough and this is most likely already being done at other places in the project.